### DISCUSSION TITLE
`watch` function in `useForm` isn't memorised

### DISCUSSION THREAD

**Discussion by RebeccaStevens** [react-hook-form/react-hook-form#3271] [Q&A] [answered]
The `watch` function returned from `useForm` is not wrapped in a `React.useCallback`. Is there any reason why it isn't by `reset`, `clearErrors` and `setError` are?

https://github.com/react-hook-form/react-hook-form/blob/edebd70abebb3ba3ce53195b265354616202d655/src/useForm.ts#L1273-L1302

**Comment by bluebill1049** [comment, marked as answer]
we don't memorize what's returned from `watch`. watch's result is meant to use in `render`

**Reply by RebeccaStevens** [reply]
I'm not talking about what's returned from `watch`, I'm referring to the function itself.

**Reply by RebeccaStevens** [reply]
Currently `watch` cannot be given (effectively) as a dependency in a dependency list as it keeps changing on every render.

**Reply by bluebill1049** [reply]
what's the use case to put `watch` in dep?

**Reply by RebeccaStevens** [reply]
I was in the middle of making a code sandbox to demonstrate but in the process I figured out how I can remove `watch` from my dependency list (by making the call to `watch` in the parent component).

If you are still curious I can finish the sandbox and link it, otherwise I think I've got a solution to my particular issue.

**Reply by bluebill1049** [reply]
thanks, @RebeccaStevens `watch` is not meant to be in the dep list, it's supposed to run before `render`.

**Reply by RebeccaStevens** [reply]
>I figured out how I can remove `watch` from my dependency list

I actually got that wrong - kinda.

`watch` was never in a dependency list, it was being passed as prop to a child component. That child component was wrapped with `React.memo` and thus shouldn't have been re-rendering when parent re-rendered unless on of it's props changed. As `watch` was changing on every render, the child component ended up re-rendering on every render too.

But anyway, I make the parent component call `watch` and pass the value down to the child component, rather than passing `watch` itself down.
